File: Algorithms/src/nattybear/level1/Divisor.hs
module Divisor where

import Data.Bool
import Data.List

solution :: [Int] -> Int -> [Int]
solution xs d = bool xs [(-1)] $ solution' xs d == []

solution' :: [Int] -> Int -> [Int]
solution' xs d = sort $ filter (\x -> x `mod` d == 0) xs

main :: IO ()
main = do
  print $ solution [5, 9, 7, 10] 5
  print $ solution [2, 36, 1, 3] 1
  print $ solution [3, 2, 6] 10

File: nattybear/level1/나누어-떨어지는-숫자-배열/Divisor.js
function solution(xs, d) {
  const ys = go(xs, d);
  if (ys.length == 0)
    return [-1];
  else
    return ys;
}

function go(xs, d) {
  return xs 
    .filter(x => x % d == 0)
    .sort((x, y) => x - y);
}

console.log(solution([5, 9, 7, 10], 5));
console.log(solution([2, 36, 1, 3], 1));
console.log(solution([3, 2, 6], 10));
